src/main/java/org/cascadebot/cascadebot/utils/ParserUtils.java

```diff
@@ -0,0 +1,45 @@
+package org.cascadebot.cascadebot.utils;
+
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ParserUtils {
+
+    private static final Pattern TIME_PARSER = Pattern.compile("(?<hours>\\d+)h|(?<minutes>\\d+)m|(?<seconds>\\d+)s");
+
+    public static long parseTime(String input, boolean onlyAllowOnce) {
+        Matcher matcher = TIME_PARSER.matcher(input);
+
+        // These booleans represent whether the unit of time has been found yet
+        boolean hoursParsed = false;
+        boolean minutesParsed = false;
+        boolean secondsParsed = false;
+
+        // The result in milliseconds
+        long millis = 0;
+        while (matcher.find()) {
+            if (matcher.group("hours") != null) {
+                if (hoursParsed && onlyAllowOnce) {
+                    throw new IllegalArgumentException("Only one of each time unit from the list: h, m, s is allowed!");
```

Comment:
Does this really need to throw an exception? You could just use `context.getTypedMessaging()` or `.replyException()` in the worst case

Reply:
I'm pretty sure it **needs** to throw an exception like this, @binaryoverload?

Reply:
This is a utils class so it should throw an exception. The exception should be handled where the util is used